cmd/operator-sdk/bundle/generate.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	gencatalog "github.com/operator-framework/operator-sdk/internal/generate/olm-catalog"
+	"github.com/operator-framework/operator-sdk/internal/generate/olm-catalog/kustomize"
+	kbutil "github.com/operator-framework/operator-sdk/internal/util/kubebuilder"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	"github.com/blang/semver"
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+)
+
+type bundleGenerateCmd struct { //nolint:maligned
+	// Which part of bundling should be run.
+	kustomize bool
+	manifests bool
+	metadata  bool
+
+	// Common options.
+	quiet        bool
+	operatorName string
+	outputDir    string
+
+	// Generate kustomize bases options.
+	apisDir string
+
+	// Generate manifests options.
+	csvVersion    string
+	manifestsRoot string
+	crdsDir       string
+	stdout        bool
+
+	// Generate metadata options.
+	directory      string
+	channels       string
+	defaultChannel string
+	overwrite      bool
+}
```

Comment:
The above split let me think that may should have more structures. Example:

```go
type bundleGenerateCmd struct { //nolint:maligned
        ... 
        manifestOptions manifestOptions
        metadataOptions metadataOptions
        ...
}

... 
type metadataOptions struct {
         directory      string
	channels       string
	defaultChannel string
	overwrite      bool
}
```

Reply:
That would make sense if we had methods only using `metadataOptions` fields, but all methods using `channels`, `defaultChannel`, and `overwrite` also use other fields from `bundleCmd`.